ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt

```diff
@@ -0,0 +1,43 @@
+package kotlinx.coroutines.javafx
+
+import javafx.beans.value.ChangeListener
+import javafx.beans.value.ObservableValue
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.conflate
+
+/**
+ * Creates an instance of a cold [Flow] that subscribes to the given [ObservableValue] and produces
+ * its values as they change.
+ *
+ * The resulting flow is conflated, meaning that if several values arrive in a quick succession, only
+ * the last one will be produced.
+ *
+ * It produces at least one value.
+ *
+ * Since this implementation uses [ObservableValue.addListener], even if this [ObservableValue]
+ * supports lazy evaluation, eager computation will be enforced while the flow is being collected.
+ */
+@ExperimentalCoroutinesApi
+fun <T: Any> ObservableValue<T>.asFlow(): Flow<T> = callbackFlow<T> {
+    val listener = ChangeListener<T> { observable, oldValue, newValue ->
+        try {
+            offer(newValue)
+        } catch (e: CancellationException) {
+            // In case the event fires after the channel is closed
+        }
+    }
+    withContext(Dispatchers.JavaFx) {
```

Comment:
Also, could you please port [this test](https://github.com/ReactiveX/RxJavaFX/blob/a78ca7d15f7d82d201df8fafb6eba732ec17e327/src/test/java/io/reactivex/rxjavafx/RxJavaFXTest.java) to `ObservableValue<T>.asFlow()`? It looks like we are missing a decent integration test

Reply:
> Is it necessary to add and remove listener in JavaFx thread?

It turns out that JavaFX is supposed to be used in a single thread for *everything*. It did catch me, too.

> Could you please point out why?

Sure: `removeListener` and `set` are not synchronized at all, and, as seen here https://java-browser.yawk.at/org.openjfx/javafx-base/11/com/sun/javafx/binding/ExpressionHelper.java#348, and here https://java-browser.yawk.at/org.openjfx/javafx-base/11/com/sun/javafx/binding/ExpressionHelper.java#311, if `removeListener` is run somewhere other than JavaFX thread, an NPE or an OutOfRange could happen.

Likewise with `addListener`: if another `addListener` was running in the JavaFX thread, similar problems would occur.

Reply:
This version does work nicely! I replaced mine with it.

I have not yet understood how to adapt the test though. I am submitting a new version of code for review anyway because introducing a new test is independent from other changes.

Reply:
Added the integration test.